apps/tangle-dapp/data/DelegationsPayouts/usePayouts.ts

```diff
@@ -0,0 +1,288 @@
+                    }
+                  }
+                });
+
+                validatorPayoutsPromises = allRewards
+                  .map(async (reward) => {
+                    const {
+                      era,
+                      totalRewardPoints,
+                      validator,
+                      validatorRewardPoints,
+                    } = reward;
+                    const validatorLedger =
+                      await apiPromise.query.staking.ledger(validator);
+
+                    const claimedEras = validatorLedger
+                      .unwrap()
+                      .claimedRewards.map((era) =>
+                        Number(era.toString().replace(/,/g, ''))
+                      );
+
+                    if (claimedEras.includes(era)) return;
+
+                    const erasTotalReward =
+                      await apiPromise.query.staking.erasValidatorReward(era);
+
+                    if (erasTotalReward.isSome) {
+                      const validatorTotalReward =
+                        (validatorRewardPoints *
+                          Number(erasTotalReward.unwrap().toString())) /
+                        totalRewardPoints;
+
+                      if (validatorTotalReward > 0) {
+                        const validatorTotalRewardFormatted =
+                          await formatTokenBalance(
+                            new u128(
+                              apiPromise.registry,
+                              validatorTotalReward.toString()
+                            )
+                          );
+
+                        const eraStaker =
+                          await apiPromise.query.staking.erasStakers(
+                            era,
+                            validator
+                          );
+
+                        const validatorTotalStake = eraStaker.total.unwrap();
+
+                        const validatorTotalStakeFormatted =
+                          await formatTokenBalance(validatorTotalStake);
+
+                        if (
+                          Number(validatorTotalStake.toString()) > 0 &&
+                          eraStaker.others.length > 0
+                        ) {
+                          const nominatorStakeInfo = eraStaker.others.find(
+                            (nominator) => nominator.who.toString() === address
+                          );
+
+                          if (
+                            nominatorStakeInfo &&
+                            !nominatorStakeInfo.isEmpty
+                          ) {
+                            const nominatorTotalStake =
+                              nominatorStakeInfo.value.unwrap();
+
+                            if (Number(nominatorTotalStake.toString()) > 0) {
+                              const nominatorStakePercentage =
+                                (Number(nominatorTotalStake.toString()) /
+                                  Number(validatorTotalStake.toString())) *
```

Comment:
If these values are of type `BN` or a derivative, prefer using `.gt(new BN(0))`, since this conversion may fail

Reply:
Must also ensure that `validatorTotalStake` can never be `0` here. But if that's assured, then disregard this comment!